src/cript/api/exceptions.py

```diff
@@ -110,13 +110,21 @@ class CRIPTAPISaveError(CRIPTException):
     http_code: str
     api_response: str
 
-    def __init__(self, api_host_domain: str, http_code: str, api_response: str):
+    def __init__(self, api_host_domain: str, http_code: str, api_response: str, patch_request: bool, pre_saved_nodes: Optional[Set[str]] = None, json_data: Optional[str] = None):
```

Comment:
this exception class will need clean up later

Reply:
I recommend passing in the http method as a string or something, we would then manipulate the string to be uppercase so the errors are always uniform

having a bool for it works right now because we only have `POST` and `PATCH`, but in the future when we start to have `DELETE` and `PUT` this will not be flexible

I think it is fine for now to get the project done, but will need refactor in the future to be more maintainable and more friendly